mono/mini/mini-ppc.c

```diff
@@ -3706,41 +3785,76 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 			 */
 			g_assert (!cfg->method->save_lmf);
 			/*
-			 * Note: we can use ppc_r12 here because it is dead anyway:
-			 * we're leaving the method.
+			 * Note: we can use ppc_r12 (ppc_r11 on Darwin) here because it is dead
```

Comment:
i think instead of ifdef on every site where r11/r12 is referenced, it would be best to put this as a define ifdef'd on OS and use that for the simple r11 vs. r12 cases, and save ifdef for the more complex ABI differences

Reply:
Let’s perhaps see what upstream says. That was my initial idea, but then I thought it may introduce extra complications down the road.